### DISCUSSION TITLE
How do I connect from VSCODE on windows to code-server on termux

### DISCUSSION THREAD

**Discussion by bropines** [coder/code-server#6808] [Q&A] [answered]
I wanted to connect to CS from a windows client, but neither remote ssh nor tunnels are available, and I don’t want control from the browser. Is it possible to connect to CS from a client on Windows?

**Comment by lg-nineoneninethreenine** [comment, marked as answer]
You wish to use code-server on Termux from VS Code on Windows. In this case, I don't see the use of code-server. You can do this(something I have tried, and it works for my use case):
1. Install and setup OpenSSH Server on Termux.
2.  Use a VPN network like Tailscale if you can't access the Termux instance via SSH normally.(It works for me for my use case, but I don't know how it might work out for you, so sorry if this doesn't work as expected)
3. Use the Remote SSH extension to connect to your Termux instance. If you use a VPN network, use the IP address that shows up on that network.

code-server can't be directly used in VS Code. It is an IDE on the browser, so you theoretically could ignore the steps I wrote up there, and just install the PWA for code-server. Not much of a difference, but it allows for keyboard shortcut recognition with ease.

**Comment by bropines** [comment]
I just thought that it would still be possible to run Code tunnel the same way it works in linux on wsl, for example. Well, it's not fate. I need vscode on termux stupidly so as not to edit files from my phone. 

By the way, what is the problem with the original code tunnel when installing over SSH? Like there's a compilation error and it's not hard to fix, or does Microsoft not want to?

**Reply by lg-nineoneninethreenine** [reply]
I couldn't get your question, unfortunately. However, you can use code-server without any VS Code involvement. You should expose it to your WiFi Network, and add HTTPS(I have seen that HTTPS is a necessity if you use it on devices like iPad). The documentation site https://coder.com/docs/code-server should have more info regarding this.

Or you could install SSH and continue, and if you meet errors when you install, that can only be solved with help from an experienced Termux user(not me, I am dumb with Termux, treating it like my PC) to resolve the error.

And sorry for the late reply, I have been very busy and ignored this discussion. Sorry!

**Reply by bropines** [reply]
Don't worry. I will still try to fix the original code tunnel and if possible, I will come back here. Thanks for the reply.
